### PR TITLE
Bump FXIOS-5855 [v112] Glean to 52.4.0

### PR DESCRIPTION
**Pull request by lmarceau**
## [FXIOS-5855](https://mozilla-hub.atlassian.net/browse/FXIOS-5855) https://github.com/mozilla-mobile/firefox-ios/issues/13374
Bumping Glean to fix crash from this issue

### CODE REVIEW

**Comment by lmarceau** [comment]
Telemetry Unit tests are failing with this update
Screen Shot 2023-03-09 at 2 58 36 PM

**Comment by lmarceau** [comment]
Created [FXIOS-5957](https://mozilla-hub.atlassian.net/browse/FXIOS-5957) to revert `Glean.shared.resetGlean(clearStores: false)` changes to `true` once we have a fix for [Glean issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1818361). Hopefully we'll get a fix for this next week since in the meantime the store for Glean tests isn't reset (& tests don't have a clean state which isn't good). Unblocking with this temporary fix is necessary so we can have the FXIOS-5855 bugfix in v112 which will be soft frozen on Monday.

File: Tests/ClientTests/TelemetryWrapperTests.swift [hunk 1]
```diff
@@ -154,16 +154,6 @@ class TelemetryWrapperTests: XCTestCase {
...
-    func test_tabsNormalQuantityWithoutExtras_GleanIsNotCalled() {
-        TelemetryWrapper.recordEvent(category: .information, method: .background, object: .tabNormalQuantity, value: nil, extras: nil)
-        XCTAssertNil(GleanMetrics.Tabs.normalTabsQuantity.testGetValue())
-    }
-
-    func test_tabsPrivateQuantityWithoutExtras_GleanIsNotCalled() {
-        TelemetryWrapper.recordEvent(category: .information, method: .background, object: .tabPrivateQuantity, value: nil, extras: nil)
-        XCTAssertNil(GleanMetrics.Tabs.privateTabsQuantity.testGetValue())
-    }
-
```
Comment (nishant2718) [inline Tests/ClientTests/TelemetryWrapperTests.swift:166]: We don't need these at all now? Even after the incoming patch?

**Review by nishant2718 — APPROVED** [review]
Looks good! No blockers - one quick question.

File: Tests/ClientTests/TelemetryWrapperTests.swift (hunk 1 above)
Comment (lmarceau) [reply Tests/ClientTests/TelemetryWrapperTests.swift:166]: I removed them and will add them back when the bug fix with Glean happens. We don't have a clean store, so getting nil for tabs will not be possible unless we force a specific order on the tests. There's existing tests through TabsQuantityTelemetryTests for `normalTabsQuantity` and `privateTabsQuantity` in the meantime.

**Pull request merged**